python-package/lightgbm/callback.py

```diff
@@ -280,18 +280,17 @@ def __init__(
         verbose: bool = True,
         min_delta: Union[float, List[float]] = 0.0,
     ) -> None:
-        if not isinstance(stopping_rounds, int) or stopping_rounds <= 0:
-            raise ValueError(f"stopping_rounds should be an integer and greater than 0. got: {stopping_rounds}")
+        self.stopping_rounds = stopping_rounds
+
+        self.enabled = _should_enable_early_stopping(stopping_rounds)
 
         self.order = 30
         self.before_iteration = False
 
-        self.stopping_rounds = stopping_rounds
```

Comment:
Why did this line need to be moved up before `self.enabled`? If it doesn't need to be and this is just left over from an earlier version of this PR, could you please revert that change?

That'd make the diff a little bit smaller, and would ensure this function doesn't spend time doing the assignment `self.stopping_rounds = stopping_rounds` if `stopping_rounds` is not an integer and the `ValueError` is raised.

Reply:
Just a mistake, I'll fix it.